llvm_jni/dist/native/nballerina.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+set -e
+
+print_usage_and_exit() {
+    echo "Usage: $0 <filename.bal> [--target aarch64]"
+    exit "$1"
+}
+
+if [ $# -lt 1 ]; then
+    print_usage_and_exit 1
+fi
+
+if [ "$1" = "--help" ] || [ "$1" = "-h" ]; then
+    print_usage_and_exit 0
+fi
+
+scriptDir=$(dirname "$0")
+src="$1"
+buildDir="$(pwd)/build"
+rm -rf "$buildDir"
+mkdir -p "$buildDir"
+
+c_compiler=cc
+runtime="$scriptDir/./balrt.a"
+if [ $# -gt 1 ]; then
+    if [ "$2" = "--target" ]; then
+        if [ $# -lt 3 ]; then
+            print_usage_and_exit 1
+        fi
+        if [ "$3" = "aarch64" ]; then
+            "$scriptDir/./compiler" --outDir "$buildDir" --target aarch64-unknown-linux-gnu "$src"
+            c_compiler=aarch64-linux-gnu-gcc
+            runtime="$scriptDir/./balrt_aarch64.a"
+        else
+            echo "Error: unsupported target $3"
+            exit 1
+        fi
+    else
+        print_usage_and_exit 1
+    fi
+else
+    "$scriptDir"/./compiler --outDir "$buildDir" "$src"
+fi
+
+objects=$(find "$buildDir" -name "*.o" | tr '\n' ' ')
```

Comment:
This won't work if there are spaces in filenames.

Reply:
Will the .o files be at any depth?

Reply:
Fixed and limited the search depth to 1